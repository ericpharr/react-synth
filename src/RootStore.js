import MonoSynth from './models/MonoSynth'
import { types } from 'mobx-state-tree'

const RootStore = types
    .model({
        monosynth: types.optional(MonoSynth, MonoSynth.create({}))
    })

export default RootStore 