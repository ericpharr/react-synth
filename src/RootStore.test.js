import RootStore from './RootStore'

jest.mock('tone')

it('can create a RootStore', () => {
    const store = RootStore.create({monosynth: {filter: {rolloff: -12}}})
    
    expect(store).toMatchSnapshot()
})