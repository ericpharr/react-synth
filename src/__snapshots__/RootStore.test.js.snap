// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a RootStore 1`] = `
Object {
  "monosynth": Object {
    "detune": 0,
    "envelope": Object {
      "attack": 0.005,
      "decay": 0.1,
      "release": 1,
      "sustain": 0.5,
    },
    "filter": Object {
      "Q": 6,
      "rolloff": -12,
      "type": "lowpass",
    },
    "filterEnvelope": Object {
      "attack": 0.06,
      "baseFrequency": 200,
      "decay": 0.2,
      "exponent": 2,
      "octaves": 7,
      "release": 2,
      "sustain": 0.5,
    },
    "oscillator": Object {
      "type": "square",
    },
  },
}
`;
